src/main/java/org/bytedeco/javacpp/tools/TemplateMap.java

```diff
@@ -48,7 +48,7 @@ boolean empty() {
                 return false;
             }
         }
-        return !isEmpty();
+        return true;
```

Comment:
This change probably make it impossible to match something like `S<>`...

Reply:
So if you don't need to "fix" this, please revert that?

Reply:
That's the fix to prevent full specializations to be automatically instantiated, using the name of the template, generating the spurious instances.
Because if empty() returns false when the template has no parameters, in `DeclarationList.add` we don't query the infomap and take the code path that trigger the instantiatation.

Reply:
Yeah ok but `S<>` also gets parsed in that case, right?

Reply:
What do you mean ? 
With `return !isEmpty` and example code above, we will generate automatically a Java class named `S` with the content of the template specialization whatever is in infoMap. With `return true`, nothing is generated, unless perhaps we have `S<>` info. And of course if we have`S<int>` info, but using the primary template.

Reply:
> will generate automatically a Java class named `S`

Named `S<int>.java`, in fact.

Reply:
Well anyway, so let's merge this for now I guess?

Reply:
I think so. 
Solving this issue about specializations will take too much time I guess.
I'm realizing that `templateArguments` doesn't work with empty list `<>`. It returns a 1-element array containing `null`. Do you want a commit for that in this PR (1-line) ?

Reply:
Sure, please do!